modules/system/classes/MediaLibrary.php

```diff
@@ -665,17 +669,19 @@ protected function scanFolderContents($fullFolderPath)
             'folders' => []
         ];
 
-        $files = $this->getStorageDisk()->files($fullFolderPath);
-        foreach ($files as $file) {
-            if ($libraryItem = $this->initLibraryItem($file, MediaLibraryItem::TYPE_FILE)) {
-                $result['files'][] = $libraryItem;
+        $contents = $this->getStorageDisk()->listContents($fullFolderPath);
+
+        foreach ($contents as $content) {
+            if ($content['type'] === 'file') {
+                $type = MediaLibraryItem::TYPE_FILE;
+                $key = 'files';
+            } elseif ($content['type'] === 'dir') {
+                $type = MediaLibraryItem::TYPE_FOLDER;
+                $key = 'folders';
             }
-        }
 
-        $folders = $this->getStorageDisk()->directories($fullFolderPath);
-        foreach ($folders as $folder) {
-            if ($libraryItem = $this->initLibraryItem($folder, MediaLibraryItem::TYPE_FOLDER)) {
-                $result['folders'][] = $libraryItem;
+            if ($libraryItem = $this->initLibraryItem($content, $type)) {
```

Comment:
@DieterHolvoet I usually prefer variable assignments to happen before they are used as a condition, as it allows tools like Xdebug to review the variable before it jumps around in the condition statement. (I realise this wasn't your doing - but let's fix it anyways).

```suggestion
            $libraryItem = $this->initLibraryItem($content, $type);
            if (!is_null($libraryItem)) {
```